server/channels/app/session.go

```diff
@@ -133,6 +136,17 @@ func (a *App) GetSessions(c request.CTX, userID string) ([]*model.Session, *mode
 		return nil, model.NewAppError("GetSessions", "app.session.get_sessions.app_error", nil, "", http.StatusInternalServerError).Wrap(err)
 	}
 
+	// MM-55320: the problem MaxSessionsLimit prevents is the OOM caused by marshalling all the sessions,
+	//  so limit sessions only when retrieving all sessions.
+	for len(sessions) > MaxSessionsLimit {
+		// Sessions are ordered by LastActivityAt DESC, so remove oldest
+		toRevoke := sessions[len(sessions)-1]
+		if err := a.RevokeSession(c, toRevoke); err != nil {
+			return nil, model.NewAppError("GetSessions", "app.session.get_sessions.app_error", nil, "", http.StatusInternalServerError).Wrap(err)
+		}
+		sessions = sessions[:len(sessions)-1]
```

Comment:
We also need to set the slice entry to nil explicitly for it to be GCd

Reply:
That's only needed if we hang on to the sessions slice, right?